### ISSUE TITLE
lerna to update dependencies of updated packages from "*" to newer version

### ISSUE THREAD

**Issue by veeramarni** [lerna/lerna#1510]
This is a continuation to request to #1486

We have a dedicated a branch called `publish` for publishing and tagging of the packages. The reason was to have our base repo to be used as a template for other multiple projects and we don't want to carry on the versions. Again, you may see problems with this but it was working very well for us.

The problem we having with our approach is when we merge master with publish and if the package's dependencies of other packages were overwritten from master to default version (which is `*`), those version won't be updated by `lerna publish`

## Expected Behavior

```
{
  "name": "my-application/sub-tool",
  "version": "0.0.15-63",
  "dependencies": {
    "@my-namespace/my-tool": "*"
  }
}
```
should be changed to

```
{
  "name": "my-application/sub-tool",
  "version": "0.0.15-63",
  "dependencies": {
    "@my-namespace/my-tool": "^0.0.15-63"
  }
}


```
## Current Behavior

```
{
  "name": "my-application/sub-tool",
  "version": "0.0.15-63",
  "dependencies": {
    "@my-namespace/my-tool": "*"
  }
}
```

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.11.0",
  "commands": {},
  "packages": [
    "packages/*",
  ],
  "version": "0.0.15-63"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.11.0 |
| `npm --version`  | 6.1.0 |
| `node --version` | 8.9.4 |

| OS | Version |
| --- | --- |
| NAME | VERSION |
| macOS Sierra | 10.12.3 |

In summary, we would like to have `lerna` automatically update the dependencies listed as "*" of updated packages to the latest release inaddition to what it is doing now with `^(carat)` (as mentioned in [here](https://github.com/lerna/lerna/blob/master/commands/publish/README.md#usage) point 4. Or we need some pointers how we can make this.

**Comment by evocateur** [comment]
You should just upgrade to lerna 3 and use relative file specifiers.

**Comment by evocateur** [comment]
Using `*` as a dependency range is not generally recommended for npm packages (not specific to lerna). One of the assumptions lerna makes is that you’re using the default ^ range, and it always updates bumped dependency ranges during publish.

Star ranges for local sibling dependencies are not explicitly supported by lerna. That they *have* worked at all is largely an accident.

If you use relative file specifiers, `lerna bootstrap` is irrelevant. Just `npm ci` and you’re done. Lerna 3 “traditional” bootstrap with lockfiles in each leaf package doesn’t change the guidance.

**Comment by veeramarni** [comment]
Well, if I change `*` to `^0.0.1`,  `lerna` still doesn't update the dependencies. I will try a quick test with Lerna 3 and see if it helps.

```
{
  "name": "my-application/sub-tool",
  "version": "0.0.15-63",
  "dependencies": {
    "@my-namespace/my-tool": "^1.0.0"
  }
}
```

**Comment by veeramarni** [comment]
@evocateur thanks for your help.
Changing to `relative` path with lerna v3 was working as expected. I think this is what I was looking for, so we don't need to carry any versions of our base project.


```
{
  "name": "my-application/sub-tool",
  "version": "0.0.15-63",
  "dependencies": {
    "@my-namespace/my-tool": "file:../my-tool"
  }
}
```
